sound/soc/sof/topology.c

```diff
@@ -465,73 +465,67 @@ static enum sof_comp_type find_process_comp_type(enum sof_ipc_process_type type)
 	return SOF_COMP_NONE;
 }
 
-/*
- * Topology Token Parsing.
- * New tokens should be added to headers and parsing tables below.
- */
-
-struct sof_topology_token {
-	u32 token;
-	u32 type;
-	int (*get_token)(void *elem, void *object, u32 offset, u32 size);
-	u32 offset;
-	u32 size;
-};
-
-static int get_token_u32(void *elem, void *object, u32 offset, u32 size)
+u32 get_token_u32(void *elem, void *object, u32 offset, u32 size)
 {
 	struct snd_soc_tplg_vendor_value_elem *velem = elem;
 	u32 *val = (u32 *)((u8 *)object + offset);
 
 	*val = le32_to_cpu(velem->value);
-	return 0;
+	return *val;
 }
 
-static int get_token_u16(void *elem, void *object, u32 offset, u32 size)
+static u32 get_token_u16(void *elem, void *object, u32 offset, u32 size)
 {
 	struct snd_soc_tplg_vendor_value_elem *velem = elem;
 	u16 *val = (u16 *)((u8 *)object + offset);
 
 	*val = (u16)le32_to_cpu(velem->value);
-	return 0;
+	return *val;
 }
 
-static int get_token_uuid(void *elem, void *object, u32 offset, u32 size)
+static u32 get_token_uuid(void *elem, void *object, u32 offset, u32 size)
 {
 	struct snd_soc_tplg_vendor_uuid_elem *velem = elem;
 	u8 *dst = (u8 *)object + offset;
 
 	memcpy(dst, velem->uuid, UUID_SIZE);
 
-	return 0;
+	return *dst;
```

Comment:
The original `int` return was meant to support failure, no?
I don't see why changing it to `u32` is good and how one should interpret the first 32bits of a UUID as `u32` as return value...

Reply:
@ujfalusi UUID is a special case. we'll have to find a way to handle it differently. This PR is not complete yet, it is only a reference. @bardliao is working on extending it for all cases. I will close this one once he posts his

Reply:
as for returning u32, I need the value from find_format() , so I had to return it and couldnt do it for just one token as it is an op. Ideally we should add a new op say get_token_new to all tokens and eventually replace get_token(). Again something @bardliao is working on.

Reply:
@ranj063, what was wrong with the pointer+offset+size? If you don't want it to be put into the ipc struct, then:
```c
struct snd_sof_tuple my_tuple;

if (token->type == SND_SOC_TPLG_TUPLE_TYPE_STRING) {
	my_tuple.token = token->token;
	token->get_token(elem, &my_tuple.value, 0, sizeof(my_tuple.value));
}
```